src/controllers/hooks/router-store/index.tsx

```diff
@@ -18,3 +29,58 @@ export const useRouterActions = (): RouterActionsType => {
 
   return allActions;
 };
+
+const createQueryParamHook = createHook<
+  EntireRouterState,
+  AllRouterActions,
+  string,
+  { paramKey: string }
+>(RouterStore, {
+  selector: ({ query }, { paramKey }): string => query[paramKey],
+});
+
+const createPathParamHook = createHook<
+  EntireRouterState,
+  AllRouterActions,
+  string,
+  { paramKey: string }
+>(RouterStore, {
+  selector: ({ match: { params } }, { paramKey }): string =>
+    params[paramKey] as string,
+});
+
+/**
+ * Utility hook for accessing URL query params
+ */
+export const useQueryParam = (
+  paramKey: string
+): [string | undefined, (qp: string | null | undefined) => void] => {
+  const [paramVal, routerActions] = createQueryParamHook({ paramKey });
+
+  const setQueryParam = React.useCallback(
+    (newValue: string | undefined | null) => {
+      routerActions.pushQueryParam({ [paramKey]: newValue });
+    },
+    [paramKey, routerActions]
+  );
+
+  return [paramVal, setQueryParam];
+};
+
+/**
+ * Utility hook for accessing URL path params
+ */
+export const usePathParam = (
+  paramKey: string
+): [string | undefined, (pp: string) => void] => {
+  const [paramVal, routerActions] = createPathParamHook({ paramKey });
+
+  const setPathParam = React.useCallback(
+    (newValue: string) => {
```

Comment:
We should support also `undefined`, as path can have optional values

Reply:
https://github.com/atlassian-labs/react-resource-router/pull/42#discussion_r492471177